core/state/statedb.go

```diff
@@ -616,7 +616,13 @@ func (s *StateDB) CommitTo(dbw trie.DatabaseWriter, deleteEmptyObjects bool) (ro
 		delete(s.stateObjectsDirty, addr)
 	}
 	// Write trie changes.
+	s.db.SetBlockNumber(blockNumber)
```

Comment:
Wouldn't it be cleaner to pass `blockNumber` as a parameter to `CommitTo` instead of having that rely on a pre-set field?